src/entry-client.js

```diff
@@ -1,4 +1,9 @@
-import { find, isString } from 'lodash-es';
+import { find, isFunction, isString } from 'lodash-es';
+
+const shouldIgnoreRouteUpdate = (c, args) => (
+    isFunction(c.shouldIgnoreRouteUpdate) &&
+    c.shouldIgnoreRouteUpdate(args) === true
```

Comment:
To ensure backwards compatibility, don't ignore if a component has not implemented the function